### ISSUE TITLE
serving packets via BPF

### ISSUE THREAD

**Issue by jcw** [iovisor/bcc#3372]
I'm at my wits' end, posting here in the hope that someone will point me in the right direction.

* I'm trying to debug an OSS TCP stack I am writing for embedded microcontroller use, written from scratch in C++
* my plan is to debug on macOS (or Linux), using BPF to read/write raw packets, as µC uploads slow down dev
* I have the code hooked up to `en0`, it reads ethernet packets and it responds to them (as seen in WireShark)
* but despite responding to SYN with a SYN+ACK, _something_ seems to be injecting a RST packet as well
* as a result, the connection fails, with "connection refused" on the `netcat` command I'm testing with

So basically, I'm trying to use BPF to debug and test a minimal TCP server in user space, via the `/dev/bpf*` device.
Is this a bad idea? Is the RST inherent in the fact that the Mac is not running a "real" server? (the firewall is turned off)

With my apologies for posting here - feel free to close if this is misplaced, but a quick hint or pointer would be appreciated ...
-jcw

UPDATE - It looks like this could be solved by turning the firewall back _on_, as BPF is in front of it.

**Comment by jcw** [comment]
Solved: firewall on, using second machine, i.e. external LAN interface, then I can send & receive.
